lib/index.js

```diff
@@ -32,9 +28,9 @@ module.exports = (opts = {}) => {
         }
 
         try {
-            if(isFunction(secret)) {
-              // secret is RSA callback (e.g. jwks-rsa.expressJwtSecret)
-              secret = await getRSASecret(ctx, secret, token);
+            if(typeof secret === 'function') {
+              // secret is RSA provider
```

Comment:
fixed, also renamed jwks.js to get-secret.js and pass whole header to the secret provider to make it more general, that will be destructured to {alg, kid} in jwks handler.